test/unit/clarity_ext/dilution/test_dilutions.py

```diff
@@ -0,0 +1,74 @@
+import unittest
+from mock import MagicMock
+from clarity_ext.dilution import DilutionScheme
+from test.unit.clarity_ext.helpers import fake_analyte
+
+
+class TestDilutionScheme(unittest.TestCase):
+    """
+    Validates that the DilutionScheme property of the context object returns the expected values
+    based on fake data from the LIMS
+    """
+
+    def test_dilution_scheme_hamilton_expected_results(self):
+        """Dilution scheme created by mocked analytes is correctly generated for Hamilton"""
+        # Setup:
+        repo = MagicMock()
+        repo.all_analytes = all_analyte_test_set1
+        dilution_scheme = DilutionScheme(repo, "Hamilton")
+
+        expected = [
+            ['art-name1', '36', 'DNA1', '14.9', '5.1', '34', 'END1'],
+            ['art-name2', '33', 'DNA2', '14.9', '5.1', '17', 'END2'],
+            ['art-name3', '50', 'DNA2', '14.9', '5.1', '44', 'END1'],
+            ['art-name4', '93', 'DNA2', '14.9', '5.1', '69', 'END2']]
+
```

Comment:
Yes, I agree. It's difficult to find that level, I think.

In this case, I feel it's doing the right thing. I think of it this way: Here we are only doing *one thing* that actually does something, i.e. setting up the object and then viewing it through the `dilutes` attribute. That I think should correspond to one test, since otherwise, one is doing that same action over and over again. I'm not so much concerned about the performance implications of that, but rather the maintainability.

I would though argue that it should have more asserts (but in one test), asserting each fact that should be true separately. Then, when the test fails in the future, it would be clear which part of the test failed. I think that would be excellent in a refactoring session.